client/src/main/java/io/confluent/kafka/schemaregistry/client/LocalSchemaRegistryClient.java

```diff
@@ -156,4 +163,50 @@ public synchronized int getVersion(String subject, Schema schema)
       throw new IOException("Cannot get version from schema registry!");
     }
   }
+  
+  @Override
+  public boolean testCompatibility(String subject, Schema newSchema) throws IOException,
+      RestClientException {
+    SchemaMetadata latestSchemaMetadata = getLatestSchemaMetadata(subject);
+    Schema latestSchema = getSchemaByIdFromRegistry(latestSchemaMetadata.getId());
+    String config = configCache.getOrDefault(subject, DEFAULT_COMPATIBILITY);
+    SchemaValidator validator = null;
+    if ("FULL".equals(config)) {
+      validator = new SchemaValidatorBuilder().mutualReadStrategy().validateLatest();
+    } else if ("FORWARD".equals(config)) {
+      validator = new SchemaValidatorBuilder().canBeReadStrategy().validateLatest();
+    } else if ("BACKWARD".equals(config)) {
+      validator = new SchemaValidatorBuilder().canReadStrategy().validateLatest();
+    } else if ("NONE".equals(config)) {
+      validator = new SchemaValidator() {
+        @Override
+        public void validate(Schema schema, Iterable<Schema> schemas)
+            throws SchemaValidationException {
+          // do nothing
+        }
+      };
+    }
+
+    List<Schema> schemas = new ArrayList<Schema>();
+    schemas.add(latestSchema);
+
+    try {
+      validator.validate(newSchema, schemas);
+    } catch (SchemaValidationException e) {
+      return false;
+    }
+    return true;
+  }
```

Comment:
It would be simpler to call AvroCompatibilityLevel.forName(config) and then call compatibilityChecker.isCompatible().

Also, could we rename config to compatibility?